hil-test/tests/ecc.rs

```diff
@@ -0,0 +1,1046 @@
+                        let modulus = DynResidueParams::new(&U256::from_be_slice(prime_field));
+                        let x_affine = DynResidue::new(
+                            &U256::from_be_slice(q.x().unwrap().as_slice()),
+                            modulus,
+                        );
+                        let y_affine = DynResidue::new(
+                            &U256::from_be_slice(q.y().unwrap().as_slice()),
+                            modulus,
+                        );
+                        let z = DynResidue::new(&U256::from_be_slice(k), modulus);
+                        let x_jacobian = x_affine * z * z;
+                        let y_jacobian = y_affine * z * z * z;
+                        sw_x.copy_from_slice(x_jacobian.retrieve().to_be_bytes().as_slice());
+                        sw_y.copy_from_slice(y_jacobian.retrieve().to_be_bytes().as_slice());
+                    }
+                    _ => unimplemented!(),
+                };
+
+                for (a, b) in x.iter().zip(sw_x.iter()) {
+                    assert_eq!(a, b);
+                }
+
+                for (a, b) in y.iter().zip(sw_y.iter()) {
+                    assert_eq!(a, b);
+                }
+            }
+        }
+    }
+
+    #[test]
+    #[cfg(feature = "esp32c2")]
+    fn test_ecc_finite_field_division(mut ctx: Context<'static>) {
+        for &prime_field in TEST_PARAMS_VECTOR.prime_fields {
+            let t1 = &mut [0_u8; 64];
+            let (k, y) = t1.split_at_mut(prime_field.len());
+            let (y, _) = y.split_at_mut(prime_field.len());
+            for _ in 0..TEST_PARAMS_VECTOR.nb_loop_inv {
+                loop {
+                    ctx.rng.read(k);
+                    ctx.rng.read(y);
+                    let is_zero = k.iter().all(|&elt| elt == 0) || y.iter().all(|&elt| elt == 0);
+                    let is_modulus = k.iter().zip(prime_field).all(|(&a, &b)| a == b)
+                        || y.iter().zip(prime_field).all(|(&a, &b)| a == b);
+                    if is_zero == false && is_modulus == false {
+                        break;
+                    }
+                }
+                let t2 = &mut [0_u8; 96];
+                let (sw_y, sw_k) = t2.split_at_mut(prime_field.len());
+                let (sw_k, sw_res) = sw_k.split_at_mut(prime_field.len());
+                let (sw_res, _) = sw_res.split_at_mut(prime_field.len());
+                sw_y.copy_from_slice(y);
+                sw_k.copy_from_slice(k);
+                let curve = match prime_field.len() {
+                    24 => &EllipticCurve::P192,
+                    32 => &EllipticCurve::P256,
+                    _ => unimplemented!(),
+                };
+
+                ctx.ecc.finite_field_division(curve, k, y)
+                    .expect("Inputs data doesn't match the key length selected.");
+                let end_time = SystemTimer::now();
+                delta_time += end_time - begin_time;
```

Comment:
```suggestion
                ctx.ecc
                    .finite_field_division(curve, k, y)
                    .expect("Inputs data doesn't match the key length selected.");
```
We are not using the times, and it will fail to build since we dont import `SysTimer`

Reply:
Oh, copy-paste :D Thanks!